### DISCUSSION TITLE
Setting session data globally

### DISCUSSION THREAD

**Discussion by azayarni** [blitz-js/blitz#1688] [Q&A]
Hello, I'm missing some kind of an entry point to be able to add some global data to the session context. The documentation explains how to modify PublicData in mutations https://blitzjs.com/docs/session-management#customize-session-public-data but how to make additional session parameters be available on every request.
HTTP Middleware seems to be not the right approach since it is a custom business logic use case.

**Comment by flybayer** [comment]
Hey @azayarni can you give an example or more details on exactly what you are trying to accomplish?

**Reply by azayarni** [reply]
Sure. An organization id should be added to the session context based on the current subdomain and a db query. Like foo.example.org => query orgs for "foo" => set session.orgId = org.id.

**Reply by flybayer** [reply]
Ok gotcha ya. So I'm guessing a user logs in once but then can directly access different subdomains if they have access?

In that case I think middleware is probably your best option. Session is managed through middleware so it sort of makes sense to add a middleware for this.

**Reply by azayarni** [reply]
Ok, got it. Thanks, Brandon. I have a follow-up question then. Where to put custom middlewares if you need access to Prisma etc? In the documentation, there is only a very basic example with an inline JS function.

**Reply by flybayer** [reply]
It will need to go in `blitz.config.js`. So no Typescript until I get #1325 merged.

But you can import the prisma client directly in `blitz.config.js` and do `const db = new PrismaClient()`. 

Feel free to keep asking questions :)

**Reply by azayarni** [reply]
Thanks a lot. TypeScript support for the config file was the missing brick ;) Looking forward to the merge. 👍

**Reply by flybayer** [reply]
It'd be great if you can share what you come up with for others who are curious :)
